### PR TITLE
Add support to send form data as url-encoded values

### PR DESCRIPTION
**Pull request by minudika**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 1]
```diff
@@ -439,9 +455,9 @@
      */
     @Override
     public Class[] getSupportedInputEventClasses() {
-        return new Class[] {String.class};
+        return new Class[] {String.class, Object.class};
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: I think it should be Map. Can you check?

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 2]
```diff
@@ -450,9 +466,9 @@
      */
     @Override
     public String[] getSupportedDynamicOptions() {
-        return new String[] {HttpConstants.HEADERS, HttpConstants.METHOD};
+        return new String[] {HttpConstants.HEADERS, HttpConstants.METHOD, HttpConstants.CONTENT_TYPE};
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: HttpConstants.CONTENT_TYPE should be removed right ?

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 3]
```diff
@@ -741,4 +766,28 @@ private HTTPCarbonMessage generateCarbonMessage(List<Header> headers, String con
         httpHeaders.set(HttpConstants.HTTP_METHOD, httpMethod);
         return cMessage;
     }
+
+    private String getPayload(Object payload) {
+        if (HttpConstants.MAP_KEYVALUE.equals(mapType)) {
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: need to check if the content type is  
APPLICATION_URL_ENCODED

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 4]
```diff
@@ -127,6 +134,14 @@
...
                         optional = true,
                         defaultValue = " "),
+                @Parameter(
+                        name = "encode.payload",
+                        description = "This defines whether the body of the payload of http message should be encoded" +
+                                " or not. This is useful when sending form data as url encoded values." +
+                                " Note that this can be used only when the mapper type is keyvalue.",
+                        type = {DataType.BOOL},
+                        optional = true,
+                        defaultValue = "false"),
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: Shall we remove this property and always encode the keyvalue ?

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java [hunk 5]
```diff
@@ -741,4 +766,28 @@ private HTTPCarbonMessage generateCarbonMessage(List<Header> headers, String con
...
+        if (HttpConstants.MAP_KEYVALUE.equals(mapType)) {
+            Map<String, Object> params = (HashMap) payload;
+            if (encodePayload) {
+                return params.entrySet().stream()
+                        .map(p -> encodeParameter(p.getKey()) + "=" + encodeParameter(p.getValue()))
+                        .reduce("", (p1, p2) -> p1 + "&" + p2);
+            } else {
+                return params.entrySet().stream()
+                        .map(p -> p.getKey() + "=" + p.getValue())
+                        .reduce("", (p1, p2) -> p1 + "&" + p2);
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: I dont think we need to support this case

Comment (minudika) [reply component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: Changed by 92d9c4a7865aedf2fcfbe350231a90b7ed67c4ca

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java (hunk 4 above)
Comment (minudika) [reply component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java]: +1 
Done by 92d9c4a7865aedf2fcfbe350231a90b7ed67c4ca

**Pull request merged**